fe/fe-core/src/main/java/com/starrocks/lake/Utils.java

```diff
@@ -187,6 +190,113 @@ public static void publishVersion(@NotNull List<Tablet> tablets, TxnInfoPB txnIn
+                    warehouseManager.getBackgroundWarehouse().getName());
+            warehouseId = warehouseManager.getBackgroundWarehouse().getId();
+        }
+
+        List<Long> rebuildPindexTabletIds = new ArrayList<>();
+        boolean success = 
+                processTablets(tablets, warehouseId, warehouseManager, nodeToTablets, rebuildPindexTabletIds, baseVersion);
+        if (!success && warehouseId != warehouseManager.getBackgroundWarehouse().getId()) {
+            warehouseId = warehouseManager.getBackgroundWarehouse().getId();
+            rebuildPindexTabletIds.clear();
+            nodeToTablets.clear();
+            success = 
+                    processTablets(tablets, warehouseId, warehouseManager, nodeToTablets, rebuildPindexTabletIds, baseVersion);
+        }
+        if (!success) {
+            throw new NoAliveBackendException("No alive node for handle publish version request in background warehouse");
+        }
+
+        // choose one aggregator
+        LakeAggregator lakeAggregator = new LakeAggregator();
+        ComputeNode aggregatorNode = lakeAggregator.chooseAggregatorNode(warehouseId);
+        if (aggregatorNode == null) {
+            throw new NoAliveBackendException("No alive compute node for handle aggregate publish version");
+        }
+        AggregatePublishVersionRequest request = new AggregatePublishVersionRequest();
+        List<ComputeNodePB> computeNodes = new ArrayList<>();
+        List<PublishVersionRequest> publishReqs = new ArrayList<>();
+        for (Map.Entry<ComputeNode, List<Long>> entry : nodeToTablets.entrySet()) {
+            PublishVersionRequest singleReq = new PublishVersionRequest();
+            singleReq.setBaseVersion(baseVersion);
+            singleReq.setNewVersion(newVersion);
+            singleReq.setTabletIds(entry.getValue());
+            singleReq.setTimeoutMs(LakeService.TIMEOUT_PUBLISH_VERSION);
+            singleReq.setTxnInfos(txnInfos);
+            singleReq.setEnableAggregatePublish(true);
+
+            if (!rebuildPindexTabletIds.isEmpty()) {
+                singleReq.setRebuildPindexTabletIds(rebuildPindexTabletIds);
+            }
+    
+            ComputeNodePB computeNodePB = new ComputeNodePB();
+            computeNodePB.setHost(entry.getKey().getHost());
+            computeNodePB.setBrpcPort(entry.getKey().getBrpcPort());
+    
+            computeNodes.add(computeNodePB);
+            publishReqs.add(singleReq);
+        }
+        request.setComputeNodes(computeNodes);
+        request.setPublishReqs(publishReqs);
+
+        LakeService lakeService = BrpcProxy.getLakeService(aggregatorNode.getHost(), aggregatorNode.getBrpcPort());
+        Future<PublishVersionResponse> future = lakeService.aggregatePublishVersion(request);
+
+        try {
+            PublishVersionResponse response = future.get();
+            if (response != null && response.failedTablets != null && !response.failedTablets.isEmpty()) {
```

Comment:
We should handle `status` in `PublishVersionResponse`

Reply:
> We should handle `status` in `PublishVersionResponse`

Fixed.